src/Microsoft.AspNetCore.Http.Connections/HttpConnectionDispatcher.cs

```diff
@@ -450,7 +450,17 @@ private static bool ServerHasWebSockets(IFeatureCollection features)
             }
 
             var pipeWriterStream = new PipeWriterStream(connection.Application.Output);
-            await context.Request.Body.CopyToAsync(pipeWriterStream);
+
+            await connection.Lock.WaitAsync();
```

Comment:
I'm wondering if we should expose a writer lock so this doesn't overlap with the locking for polls.

Reply:
Yes, lets do that, lets have a SendLock and ReceiveLock

Reply:
What about actions like start and stop? Would they lock both SendLock and ReceiveLock?

Reply:
Stop and Start?

Reply:
Meh locking is fine it keeps dispose simple